src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs

```diff
@@ -1132,14 +1132,26 @@ private void CheckExpectedType(JsonTokenType expected, JsonTokenType actual)
         }
 
         private static void CheckSupportedOptions(
-            JsonReaderOptions readerOptions,
+            JsonDocumentOptions options,
             string paramName = null)
         {
-            if (readerOptions.CommentHandling == JsonCommentHandling.Allow)
+            if (options.CommentHandling == JsonCommentHandling.Allow)
             {
                 throw new ArgumentException(
                     SR.JsonDocumentDoesNotSupportComments,
-                    paramName ?? nameof(readerOptions));
+                    paramName ?? nameof(options));
+            }
+        }
+
+        private static void CheckSupportedOptions(
+            JsonReaderOptions options,
+            string paramName = null)
+        {
+            if (options.CommentHandling == JsonCommentHandling.Allow)
```

Comment:
I wonder if this is better served on the options property setter (now that it's a type dedicated to the document)

Reply:
(Though, I guess this copy is for an already open reader... :smile:)